### ISSUE TITLE
using detector on (large) image crashes python with no exception...

### ISSUE THREAD

**Issue by mazuzu** [davisking/dlib#2132]
I've searched online for anyone with similar issues but I've not seen it.  I suspect this is a memory issue but it's a bit odd.

My code, based on the examples from the website, works fine on a smaller image but on the size I would like to use it crashes python.  There is no exception raised.  I'm running windows, and when I execute the code, the line highlighted below is where it stops.  Windows pops up an error box saying that python experienced an error and is closing...

The jpg file for the full image is 49MB.

## Expected Behavior
python code: 
gray = dlib.load_rgb_image("8192x8192image.jpg")
detector = dlib.simple_object_detector("custom_detector.svm")
circles = detector(gray)  # fails here on big image

this should detect many many circles in the image 'gray'

This code works just fine with a cropped piece of the image gray, but the full image is a standard output of another tool so ideally it should also work on this image.

## Current Behavior
There is no error message in the console.  Python crashes with a standard windows information box stating that python experienced an error and is closing.

## Steps to Reproduce
The 8192x8192 pixel image has, roughly, 80,000 "dots" arranged in a grid/array.  Each of these should be detected individually.   The custom detector (and also predictor) was used on multiple images of single dots (there is variation in the dots' "circularity") and a very small grid.

Again, the code works fine on a smaller image (~2300 x 2300 pixels) where it detects roughly 4000 dots.

I cannot provide the exact jpg file.  I can try to create a facsimile if necessary for testing. 

* **Version**: dlib v19.20
* **Where did you get dlib**:  installed via anaconda navigator / official release
* **Platform**: Windows 10, 64bit, 16gb memory
* **Compiler**: python 3.7

I suppose having an method to programmatically crop the image and perform the detection on that cropped image that would also suit my needs but the error on the big image would remain so I would still report it as a bug of some sort...

**Comment by davisking** [comment]
You are probably running out of RAM.  There isn't much I can do.  The C++ runtime will throw an exception, and normally you would get an exception in python.  But if you get unlucky and python can't allocate memory for whatever it happens to be doing after that then it might crash.  I also wager you have a simple object detector that's upsampling the image, so it's running on a really really big image.

**Comment by mazuzu** [comment]
I am using a detector with upsampling, although I do cap it at 3 when it was created.  I was using crops of individual dots which are images with a pixel dimension of roughly 40x40 and the training gave an error about the size until I allowed the upsample limit.  I have put in the programmatic crop code and I can step through 4 images of 4096x4096 without the crash.  I did not test what the limit was since this was reasonable enough...
